integration-test/src/test/resources/test_python_api.py

```diff
@@ -0,0 +1,186 @@
+
+    process_job(simple_spark_job, 3)
+
+
+def test_add_file():
+    add_file_name = os.path.basename(add_file_url)
+    json_data = json.dumps({'uri': add_file_url})
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/add-file"
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, data=json_data)
+
+    assert response.status_code == httplib.OK
+
+    def add_file_job(context):
+        from pyspark import SparkFiles
+        with open(SparkFiles.get(add_file_name)) as testFile:
+            file_val = testFile.readline()
+        return file_val
+
+    process_job(add_file_job, "hello from addfile")
+
+
+def test_add_pyfile():
+    add_pyfile_name_with_ext = os.path.basename(add_pyfile_url)
+    add_pyfile_name = add_pyfile_name_with_ext.rsplit('.', 1)[0]
+    json_data = json.dumps({'uri': add_pyfile_url})
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/add-pyfile"
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    response_add_pyfile = requests.request('POST', request_url, headers=header, data=json_data)
+
+    assert response_add_pyfile.status_code == httplib.OK
+
+    def add_pyfile_job(context):
+       pyfile_module = __import__ (add_pyfile_name)
+       return pyfile_module.test_add_pyfile()
+
+    process_job(add_pyfile_job, "hello from addpyfile")
+
+
+def test_upload_file():
+    upload_file = open(upload_file_url)
+    upload_file_name = os.path.basename(upload_file.name)
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/upload-file"
+    files = {'file': upload_file}
+    header = {'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, files=files)
+
+    assert response.status_code == httplib.OK
+
+    def upload_file_job(context):
+        from pyspark import SparkFiles
+        with open(SparkFiles.get(upload_file_name)) as testFile:
+            file_val = testFile.readline()
+        return file_val
+
+    process_job(upload_file_job, "hello from uploadfile")
+
+
+def test_upload_pyfile():
+    upload_pyfile = open(upload_pyfile_url)
+    upload_pyfile_name_with_ext = os.path.basename(upload_pyfile.name)
+    upload_pyfile_name = upload_pyfile_name_with_ext.rsplit('.', 1)[0]
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/upload-pyfile"
+    files = {'file': upload_pyfile}
+    header = {'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, files=files)
+    assert response.status_code == httplib.OK
+
+    def upload_pyfile_job(context):
+        pyfile_module = __import__ (upload_pyfile_name)
+        return pyfile_module.test_upload_pyfile()
+    process_job(upload_pyfile_job, "hello from uploadpyfile")
+
+
+if __name__ == '__main__':
+    test_dir_path = os.getcwd() + "/src"
+    value = pytest.main(test_dir_path)
```

Comment:
is there any cleanup hook in pytest? I notice that you're never stopping the session you create, so it will stay alive inside Livy and potentially prevent other tests from running.